src/coffea/dataset_tools/preprocess.py

```diff
@@ -147,12 +173,14 @@ class DatasetSpecOptional:
         dict[str, str] | list[str] | dict[str, UprootFileSpec | CoffeaFileSpecOptional]
     )
     metadata: dict[Hashable, Any] | None
+    form: str | None
```

Comment:
Can `DatasetSpecOptional` inherit from `DatasetSpec`?

Reply:
Yeah I can reorganize this, I forgot that python was all virtual inheritance.